gossipsub.go

```diff
@@ -94,6 +94,35 @@ func (gs *GossipSubRouter) RemovePeer(p peer.ID) {
 	delete(gs.control, p)
 }
 
+func (gs *GossipSubRouter) EnoughPeers(topic string, suggested int) bool {
+	// check all peers in the topic
+	tmap, ok := gs.p.topics[topic]
+	if !ok {
+		return false
+	}
+
+	fsPeers, gsPeers := 0, 0
+	// floodsub peers
+	for p := range tmap {
+		if gs.peers[p] == FloodSubID {
+			fsPeers++
+		}
+	}
+
+	// gossipsub peers
+	gsPeers = len(gs.mesh[topic])
+
+	if suggested == 0 {
+		suggested = GossipSubD
```

Comment:
I'm fine changing it, figured that we'd be better off proactively getting more peers between `GossipSubDlo` and `GossipSubD` so that if we ever drop below `Dlo` mesh peers we already have peers available for upgrade.

Yes there is a `Publish` option. Look at `WithReadiness` and `MinTopicSize` as they're used in the discovery tests.

Reply:
@vyzo going to make some hopefully end-of-the-road changes. Did you still think this was important or do you agree that `GossipSubD` instead of `GossipSubDlo` actually make more sense?

Reply:
I think we should use `GossipSubDlo` as it makes more sense.

Reply:
np